[BUGFIX] Make content element wrap work in TYPO3 CMS 13

thommyhh opened the pull request:
* refactor stdWrap hook to event listener
* remove hook registration in `ext_localconf.php`

Fixes #48

thommyhh committed:
[BUGFIX] Make content element wrap work in TYPO3 CMS 13

* refactor stdWrap hook to event listener
* remove hook registration in `ext_localconf.php`

thommyhh committed:
[BUGFIX] Make content element wrap work in TYPO3 CMS 13

* refactor stdWrap hook to event listener
* remove hook registration in `ext_localconf.php`

thommyhh committed:
[BUGFIX] Restore content object stdWrapp hook

thommyhh committed:
[BUGFIX] Restore content object stdWrapp hook

helhum commented on Classes/EventListener/ContentElementWrapListener.php:
```diff
@@ -0,0 +1,102 @@
+        $path = $event->getConfiguration()['turboFrameWrap.']['path'] ?? $this->determineRenderingPath($event->getContentObjectRenderer(), $event->getConfiguration());
+
+        $contextFactory = new TopwireContextFactory($event->getContentObjectRenderer()->getRequest());
+        $context = $contextFactory->forPath($path, $event->getContentObjectRenderer()->currentRecord);
```
the v12 issues are obviously bogus, but the v13 issues should be fixed. 
I will take care of adding a dedicated v12 phpstan config, can you take care fixing the v13 failures?

thommyhh replied:
Yes, I will look at the v13 issues. Can you commit directly to my branch for the v12 issues, so we get a clean PR?

thommyhh replied:
@helhum I fixed the tests for both v12 and v13 by providing an additional configuration for v12 that excludes the new file, that would not be compatible with v12.

I removed the `^` from the TYPO3 version in the matrix and added it to the `composer update` command instead, so now the required checks doesn't match anymore, so that need to be adjusted.

Are you ok with that approach?

helhum reviewed (changes requested):
Can you fix the v13 phpstan errors?

thommyhh committed:
[BUGFIX] Fix PHPStan tests

thommyhh committed:
[BUGFIX] Fix PHPStan tests

thommyhh committed:
[BUGFIX] Remove space after type cast

helhum committed:
Merge branch 'main' into bugfix/content-element-wrap

EvilBMP commented:
Any updates on this one?

helhum committed:
chore: Allow packages with advisories from being installed in CI

Also remove rector as it does not allow typo3 12 deps
to be installed due to its outdated depedencies

helhum committed:
feat: Use typo3-branch computation for phpstan

helhum reviewed (approved):
Thanks!
